.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+name: CI
```

Comment:
Seems like there are many python versions in use across the different jobs. If this is intentionally, adding a comment specifying an issue link or reason why a particular version is required will make someone (maybe you) reviewing this yaml file in 6 years very appreciative :)

If otherwise you don't need different versions you could pin the version using an environment variable and specifying this across the different jobs. This is just a suggestion.

```yaml
env:
  - PYTHON_VERSION: "3.10"

jobs:
  lint:
    ...
    with:
      python-version: ${{ env.PYTHON_VERSION }}
```

Even in a scenario where you need specific versions per purpose, using a variable can also help encapsulate that purpose.

```yaml
env:
   PYTHON_VERSION_ WITH_SPECIFIC_FEATURE: "3.6"
```

Reply:
Much of that range is due to inheriting a spread from before. The 3.12 for `lint` is just the current version and could probably be dropped to rely on the default provided by `ubuntu-latest`; `baseline` uses 3.6 because we still claim compatibility with that, and `documentation`'s 3.9 is slightly arbitrary, but I'm loathe to spend many cycles on it given the doc build's complexity.

Reply:
You explained it excellently just now. I Just wonder if pulling that information out of this comment thread into the action definition would be helpful for someone like myself reading the code for the first time without context. Just a light suggestion to do with what you think is best.